homeassistant/components/rfxtrx/__init__.py

```diff
@@ -245,6 +258,20 @@ def handle_receive(event):
         if device_id in device_events:
             hass.bus.fire(EVENT_RFXTRX_EVENT, event_data)
 
+        if config[CONF_AUTOMATIC_ADD]:
+            if device_id not in devices:
+                data = entry.data
+                data[CONF_DEVICES][event_data["data"]] = device_id
+
+                asyncio.run_coroutine_threadsafe(
```

Comment:
We can use `hass.add_job` instead of `asyncio.run_coroutine_threadsafe`.

Reply:
For my understanding, would there be cases where `asyncio.run_coroutine_threadsafe` is needed instead of `hass.add_job`? I got the former from the developer docs.

Reply:
If we need to wait for the result of the task, we can't use `hass.add_job`.

Reply:
Ah right, that makes sense, thanks!